modules/agent_files/templates/xorg.conf.erb

```diff
@@ -0,0 +1,48 @@
+Section "ServerLayout"
+    Identifier     "Layout0"
+    InputDevice    "Keyboard0" "CoreKeyboard"
+    InputDevice    "Mouse0" "CorePointer"
+EndSection
+
+Section "Files"
+EndSection
+
+Section "InputDevice"
+    # generated from default
+    Identifier     "Mouse0"
+    Driver         "mouse"
+    Option         "Protocol" "auto"
+    Option         "Device" "/dev/psaux"
+    Option         "Emulate3Buttons" "no"
+    Option         "ZAxisMapping" "4 5"
+EndSection
+
+Section "InputDevice"
+    # generated from default
+    Identifier     "Keyboard0"
+    Driver         "kbd"
+EndSection
+
+Section "Monitor"
+    Identifier     "Monitor0"
+    VendorName     "Unknown"
+    ModelName      "Unknown"
+    HorizSync       28.0 - 33.0
+    VertRefresh     43.0 - 72.0
+    Option         "DPMS"
+EndSection
+
+Section "Device"
+    Identifier     "Device0"
+    Driver         "nvidia"
+    VendorName     "NVIDIA Corporation"
+    BoardName      "GRID K520"
+    BusID          "<%= @facts['gpu_device_bus_id'] %>"
```

Comment:
Adding this as a fact relies on the presence of an executable not available until puppet installs the executable. I haven't found a reasonable way to populate a variable at puppet runtime. You can try using the erb directive like 

```erb
BusID    "<%= `nvidia-xconfig --query-gpu-info  | grep BusID | sed "s/.*PCI:/PCI:/g"` %>
```
which uses Ruby's ability to shell out but I feel like a sensible tool would prevent that from working.

Alternatively this is something that we set in a hiera configuration file.

Reply:
The approach of embedding nvidia-xconfig into the ERB in the form of `fact` or calling shell directly into the ERB code can not work since the evaluation is done too early and package dependencies are not resolved yet.